### ISSUE TITLE
Unable to set sidecar.istio.io Pod annotations in jaeger pod

### ISSUE THREAD

**Issue by cnvergence** [jaegertracing/helm-charts#355]
**Describe the bug**
Currently, we are unable to add annotations properly to a Jaeger pod using Jaeger-operator.

**To Reproduce**
Applying those annotations in the values of Jaeger-operator helm chart.
```bash
jaeger:
  spec:
    annotations:
      sidecar.istio.io/inject: "true"
      sidecar.istio.io/rewriteAppHTTPProbers: "true"
 ```
This will result in the following `istioctl analyze` warnings:
```bash
Warning [IST0107] (Deployment kyma-system/tracing-jaeger) Misplaced annotation: sidecar.istio.io/inject can only be applied to Pod
Warning [IST0107] (Deployment kyma-system/tracing-jaeger) Misplaced annotation: sidecar.istio.io/rewriteAppHTTPProbers can only be applied to Pod
```
I tried then to apply them to the Pods, with those values:
```bash
jaeger:
  spec:
    template:
      metadata:
        annotations:
          sidecar.istio.io/inject: "true"
          sidecar.istio.io/rewriteAppHTTPProbers: "true"
```
The result was, that it did not update those annotations in the pods, instead, it got overwritten with
` sidecar.istio.io/inject: "false"` annotation

**Expected behaviour**
It should be possible to control Pod annotations on Jaeger pods controlled by Jaeger-operator.

**Version (please complete the following information):**
 - OS: [e.g. Linux]
 - Jaeger version: 1.30.0
 - Deployment: Kubernetes with Jaeger Operator

**Comment by cnvergence** [comment]
The overwrite with `sidecar.istio.io/inject: "false"` should be fixed with this PR
- https://github.com/jaegertracing/jaeger-operator/pull/1860

**Comment by dimitarshenkov** [comment]
`jaeger:
  spec:
    template:
      metadata:
        annotations:
          sidecar.istio.io/inject: "true"`
Did not add the  sidecar.istio.io/inject: "true"`  to the pod

**Comment by cnvergence** [comment]
The problem was solved only for 1.34.0.
Adding those values was setting annotations properly
```bash
jaeger:
  spec:
    allInOne:
      annotations:
        sidecar.istio.io/inject: "true"
        sidecar.istio.io/rewriteAppHTTPProbers: "true"        
```
Reopening this issue, because from 1.35.0, it still exists.

**Comment by cnvergence** [comment]
Adding those values will setannotations properly
```bash
jaeger:
  spec:
    allInOne:
      annotations:
        sidecar.istio.io/inject: "true"
        sidecar.istio.io/rewriteAppHTTPProbers: "true"        
```
